### PR TITLE
Add capability to skip version checks

### LINKED ISSUE #1077: Add an option to skip check for a new version.
## Description

It would be nice to have an option to opt out of the version check, either with an environment variable or a `--skip-version-check` option.

This would allow skiping version checks when doing repo migrations at scale.

### PR DESCRIPTION
**Pull request by tspascoal**
If  environment variable GEI_SKIP_VERSION_CHECK is set true, no version checks will be performed

Closes #1077 

- [x] Did you write/update appropriate tests
- [x] Release notes updated (if appropriate)
- [x] Appropriate logging output
- [x] Issue linked
- [x] Docs updated (or issue created)
- [x] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/ado2gh/Program.cs [hunk 1]
```diff
@@ -106,6 +106,12 @@ private static async Task GithubStatusCheck(ServiceProvider sp)
 
         private static async Task LatestVersionCheck(ServiceProvider sp)
         {
+            if (Environment.GetEnvironmentVariable("GEI_SKIP_VERSION_CHECK")?.ToUpperInvariant() == "TRUE")
```
Comment (ArinGhazarian) [inline src/ado2gh/Program.cs:109]: same [suggestions](https://github.com/github/gh-gei/pull/1155#discussion_r1373701295) applies here too

File: src/ado2gh/Program.cs [hunk 2]
```diff
@@ -114,6 +114,14 @@ private static async Task GithubStatusCheck(ServiceProvider sp)
 
         private static async Task LatestVersionCheck(ServiceProvider sp)
         {
+            var envProvider = sp.GetRequiredService<EnvironmentVariableProvider>();
+
+            if (envProvider.SkipVersionCheck()?.ToUpperInvariant() is "TRUE" or "1")
+            {
+                Logger.LogInformation("Skipped latest version check of the ado2gh CLI due to GEI_VERSION_CHECK environment variable");
```
Comment (timrogers) [inline src/ado2gh/Program.cs:121]:
```suggestion
                Logger.LogInformation("Skipped latest version check due to GEI_VERSION_CHECK environment variable");
```

File: src/gei/Program.cs [hunk 3]
```diff
@@ -114,6 +114,14 @@ private static async Task GithubStatusCheck(ServiceProvider sp)
 
         private static async Task LatestVersionCheck(ServiceProvider sp)
         {
+            var envProvider = sp.GetRequiredService<EnvironmentVariableProvider>();
+
+            if (envProvider.SkipVersionCheck()?.ToUpperInvariant() is "TRUE" or "1")
+            {
+                Logger.LogInformation("Skipped latest version check of the gei CLI due to GEI_VERSION_CHECK environment variable");
```
Comment (timrogers) [inline src/gei/Program.cs:121]:
```suggestion
                Logger.LogInformation("Skipped latest version check due to GEI_VERSION_CHECK environment variable");
```

**Review by timrogers — APPROVED** [review]
Looks great to me! I have a few suggestions on the docs and copy, just to align with your other PR.

File: RELEASENOTES.md [hunk 4]
```diff
@@ -1 +1,3 @@
 
+- Skip last version check on startup if environment variable `GEI_SKIP_VERSION_CHECK` is set to `true`
+- Skip GitHub status checking on startup if environment variable `GEI_SKIP_STATUS_CHECK` is set to true.
```
Comment (timrogers) [inline RELEASENOTES.md:3]:
```suggestion
- Skip latest version check on startup if environment variable `GEI_SKIP_VERSION_CHECK` is set to `true`
```

**Review by timrogers — CHANGES_REQUESTED** [review]
This is looking good to me from a code perspective ✨ Can you tweak the release notes as suggested, and then we can get this merged?

**Review by timrogers — APPROVED** [review]

**Comment by timrogers** [comment]
Passed! ✅

**Pull request merged**